app/domain/content/importers/metrics.rb

```diff
@@ -0,0 +1,117 @@
+module Content
+  module Importers
+    class Metrics
+      def initialize(content_ids)
+        @content_ids = content_ids
+        @start_date = 3.months.ago.to_date
+      end
+
+      def import
+        start_time = Time.now
+        metrics = @content_ids
+          .map { |content_id| fetch_metrics(content_id) }
+          .flatten
+        fetched_metrics_end = Time.now
+        puts "Fetched all in #{fetched_metrics_end - start_time}s"
+
+        puts "Writing metrics to database..."
+        # TODO: on_duplicate_key_update
+        Content::Metric.import(metrics)
+        puts "Imported all in #{Time.now - fetched_metrics_end}s"
+      end
+
+    private
+
+      def fetch_metrics(content_id)
+        puts "Fetching metrics for #{content_id} ..."
+        versions = fetch_versions(content_id)
+
+        metrics_by_version = {}
+
+        metrics = (@start_date..Date.today).map do |date|
+          version = versions.find { |v| Date.parse(v[:updated_at]) <= date }
+
+          if version.nil?
+            nil
+          else
+            metric_for_version = metrics_by_version.fetch(version[:user_facing_version]) do |version_number|
```

Comment:
Sometimes the version is not the user facing version (it is a draft?) and so we use the reading age of the user facing version. Am I understanding that correctly?

Reply:
I'm pretty sure that historical versions have always been published (the only way of creating a new draft is by either publishing or discarding the current draft in progress). There might be funniness around the current version being a draft, but I'm not sure that's that important for this spike? Certainly something we'd have to consider for production code.